File: src/my_robot_tutorials/src/smartphone.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_receive_radio_data(const std_msgs::String&msg){
    ROS_INFO("message received: %s", msg.data.c_str());
}

int main(int argc, char**argv){
    ros::init(argc, argv, "smarphone");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/robot_news_radio", 1000, callback_receive_radio_data);

    ros::spin();

}

File: src/my_robot_tutorials/scripts/smartphone.py
#! /usr/bin/env python

# rostopic pub -1 /robot_news_radio std_msgs/String "good"

import rospy
from std_msgs.msg import String


def callback_receive_radio_data(msg):
    rospy.loginfo("message received: ")
    rospy.loginfo(msg)

if __name__=='__main__':
    rospy.init_node('smartphone')

    sub = rospy.Subscriber('/robot_news_radio', String, callback_receive_radio_data)

    rospy.spin()
